### PR TITLE
Feature/cosmoz tabs haunted

### PR DESCRIPTION
**Pull request by megheaiulian**
Done:
  - Refactor cosmoz-tabs with haunted & lit-html
  - Added storybook integration 
  - Added hash param support with `useHashParam`

### CODE REVIEW

**Comment by JockeCK** [comment]
Tested a bit, really like the story book demo setup. Something seems to go wrong though when you click on one of the tabs in the section tabs hash

**Comment by megheaiulian** [comment]
@JockeCK Ignore that problem in `hash` story for now. In FE it will work just fine. It will be improved but right now we still use `cosmoz-page-location` in some components (e.g: cosmoz-data-nav).

**Review by nomego — APPROVED** [review]
The implementation feels concise and well written!

I couldn't really digest the code on a first go so I think there's room for simplification.
Also it's risky to have wording like "elect" alongside "select" for typo, misread and just general confusion reasons. I didn't actually even analyze the purpose but maybe a word like appoint or choose if there's no better semantic angle.
Also, the useTabs hook could (should) be a bit more narrative and maybe extract some inner useEffect hooks into custom hooks with descriptive names, also making it more concise and easier to digest.

**Comment by nomego** [comment]
Oh but the tests fail so those needs fixing before merge.

**Comment by megheaiulian** [comment]
@nomego You're right about the naming so I updated `use-tabs`. Have a look!

File: lib/use-tabs.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+const useTabs = host => {
+	const {
+			selected, hashParam
+		} = host,
+		[tabs, setTabs] = useState([]),
+		param = useHashParam(hashParam),
+		selection = hashParam == null || param == null && selected != null ? selected : param,
+		selectedTab = useMemo(() => choose(tabs, selection), [tabs, selection]);
+
+	useEffect(() => {
```
Comment (nomego) [inline lib/use-tabs.js:21]:
```suggestion
	useTabSelectedEffect(selectedTab);
```
?

File: lib/use-tabs.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		return () => {
+			selectedTab.toggleAttribute('is-selected', false);
+		};
+
+	}, [selectedTab]);
+
+	useEffect(() => {
+		const onTabAlter = e => {
+			e.stopPropagation();
+			setTabs(prev => prev.slice());
```
Comment (nomego) [inline lib/use-tabs.js:46]: Based on initial state, and semantics, shouldn't tabs be an array?

**Review by nomego — APPROVED** [review]
Looks better!

File: lib/use-tabs.js (hunk 2 above)
Comment (megheaiulian) [reply lib/use-tabs.js:46]: It is a an array. The state updater can also receive a function to compute a value from previous one.

File: lib/use-tabs.js (hunk 1 above)
Comment (megheaiulian) [reply lib/use-tabs.js]: Added `useTabSelectedEffect`

**Review by cristinecula — APPROVED** [review]
Great work!

**Pull request merged**